### PR TITLE
Airflow initial release

### PR DESCRIPTION
**Pull request by ritalwar**
- Enhancement

## What does this PR do?

This PR adds Airflow experimental integration release.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
-[ ] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/main/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target are documented
- [x] Supported operating systems are documented (if applicable)
- [ ] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to:

## How to test this PR locally

Add Airflow integration in Kibana, configure airflow host and port, run the agent and check for metrics data in Kibana.

- Closes #https://github.com/elastic/integrations/issues/3748

## Screenshots

Screenshot 2022-11-13 at 11 34 36 PM
Screenshot 2022-11-13 at 11 34 06 PM

### CODE REVIEW

File: packages/airflow/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: airflow
+title: Airflow
+version: 0.0.1
+release: experimental
```
Comment (ishleenk17) [inline packages/airflow/manifest.yml:4]: The new package versioning doesn't need to have the release label. The version itself will take care of it.

File: packages/airflow/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    show_user: true
+    default: "8125"  
+policy_templates:
+  - name: statsd
+    title: Airflow metrics
+    description: Collect metrics from Airflow instances
+    data_streams:
+      - statsd
+    inputs:
+      - type: "airflow/metrics"
```
Comment (ishleenk17) [inline packages/airflow/manifest.yml:40]: We are not using statsd directly as input here ?

File: packages/airflow/data_stream/statsd/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
+- name: airflow
+  type: group
+  fields:
+    - name: '*.1m_rate'
```
Comment (ishleenk17) [inline packages/airflow/data_stream/statsd/fields/fields.yml:4]: This file has a lot of dynamic mapping for the fields. Do we know approx how many fields are being mapped here?
Can it lead to field explosion?

File: packages/airflow/_dev/build/build.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+dependencies:
+  ecs:
+    reference: git@v8.2.0
```
Comment (muthu-mps) [inline packages/airflow/_dev/build/build.yml:3]: Can we update to recent stable ECS version compatible with release version ?

File: packages/airflow/_dev/build/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+This integration collects metrics from [Airflow](https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/metrics.html) running a
+statsd server where airflow will send metrics to. The default metricset is `statsd`.
+
+## Compatibility
+
+The Airflow module is tested with Airflow 2.4.0. It should work with version
+2.0.0 and later.
+
+### statsd
```
Comment (muthu-mps) [inline packages/airflow/_dev/build/docs/README.md:11]: Please add few lines about statsd and how it collects metrics from Airflow.

File: packages/airflow/data_stream/statsd/manifest.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+type: metrics
+title: Airflow statsd metrics
+release: experimental
+streams:
+  - input: airflow/metrics
+    enabled: true
+    title: Airflow statsd metrics
```
Comment (muthu-mps) [inline packages/airflow/data_stream/statsd/manifest.yml:7]: Remove `statsd` in the title and description and include those details in README file.

File: packages/airflow/data_stream/statsd/fields/fields.yml (hunk 3 above)
Comment (ritalwar) [reply packages/airflow/data_stream/statsd/fields/fields.yml:4]: I can see the different number of fields mapped at different times on the basis of data, at max I have seen around 400 fields so far that's getting mapped.

Comment (ishleenk17) [reply packages/airflow/data_stream/statsd/fields/fields.yml:4]: I think these would be much more than the required fields. What's your thought on this? 
We should maybe choose and do only the necessary field mappings for such curated integrations.

File: packages/airflow/data_stream/statsd/agent/stream/stream.yml.hbs [hunk 7]
```diff
@@ -0,0 +1,3 @@
+metricsets: ["server"]
+host: {{host}}
+port: {{port}}
```
Comment (ishleenk17) [inline packages/airflow/data_stream/statsd/agent/stream/stream.yml.hbs:3]: Don't we need to transfer statsd_on = True variable as well to statsd through hbs file ?

File: packages/airflow/data_stream/statsd/fields/fields.yml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    - name: '*.p95'
+      type: object
+      object_type: double
+      object_type_mapping_type: "*"
+      description: Airflow 95 percentile timers metric
+    - name: '*.p99_9'
+      type: object
+      object_type: double
+      object_type_mapping_type: "*"
+      description: Airflow 99.9 percentile timers metric
```
Comment (ishleenk17) [inline packages/airflow/data_stream/statsd/fields/fields.yml:63]: What all percentile metrics do we usually include in other integrations? I see 75, 95, 99.9, and 99 percentile. 
Is it too many fields mapping?

File: packages/airflow/data_stream/statsd/manifest.yml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+type: metrics
+title: Airflow metrics
+release: experimental
+streams:
+  - input: statsd/metrics
```
Comment (ishleenk17) [inline packages/airflow/data_stream/statsd/manifest.yml:5]: Please list down the metrics that you are not receiving due to this change of input.

File: packages/airflow/manifest.yml [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+title: Airflow
+version: 0.0.1
+description: Airflow Integration.
+type: integration
+format_version: 1.0.0
+license: basic
+categories:
+  - web
+conditions:
+  kibana.version: "^8.5.0"
```
Comment (ishleenk17) [inline packages/airflow/manifest.yml:11]: Please add elastic_subscription under conditions

File: packages/airflow/data_stream/statsd/agent/stream/stream.yml.hbs (hunk 7 above)
Comment (ritalwar) [reply packages/airflow/data_stream/statsd/agent/stream/stream.yml.hbs:3]: it is enabled here: https://github.com/ritalwar/integrations/blob/migrate_airflow_3748/packages/airflow/data_stream/statsd/manifest.yml#L6

File: packages/airflow/data_stream/statsd/fields/fields.yml (hunk 3 above)
Comment (ritalwar) [reply packages/airflow/data_stream/statsd/fields/fields.yml:4]: okay, I agree, let me see what all fields we can filter out and will update accordingly.

File: packages/airflow/data_stream/statsd/manifest.yml (hunk 9 above)
Comment (ritalwar) [reply packages/airflow/data_stream/statsd/manifest.yml:5]: It is receiving same metrics as with input type as airflow.

File: packages/airflow/data_stream/statsd/fields/fields.yml (hunk 8 above)
Comment (ritalwar) [reply packages/airflow/data_stream/statsd/fields/fields.yml]: As discussed removed these percentiles metrics.

**Review by ishleenk17 — APPROVED** [review]
Looks good !

**Pull request merged**